### DISCUSSION TITLE
How do I migrate from a higher-version system to a lower-version system

### DISCUSSION THREAD

**Discussion by xiaonuoz** [cilium/ebpf#845] [Q&A] [answered]
hi!
I want to migrate my program to a lower version, but I don't know which way to achieve the LIBBPF_OPTS() effect
ps:The function is to monitor vfs read

**Comment by xiaonuoz** [comment]
c:
![image](https://user-images.githubusercontent.com/46923935/200281400-f08a3d3a-9252-410e-a148-52cf5c4339f1.png)

**Comment by florianl** [comment, marked as answer]
`LIBBPF_OPTS` is an element introduced by [libbpf](https://github.com/libbpf/libbpf). So I think your question is placed best in [this repository](https://github.com/libbpf/libbpf).

Both, [libbpf](https://github.com/libbpf/libbpf) and [cilium/ebpf](https://github.com/cilium/ebpf), are libraries helping to load eBPF into the Linux Kernel. As [libbpf](https://github.com/libbpf/libbpf) is the upstream library, that ships with the Linux Kernel itself, it includes some features and functionality that are not part (yet) of [cilium/ebpf](https://github.com/cilium/ebpf).

**Reply by ti-mo** [reply]
Agree overall, but not sure what missing feature you're alluding to here. `LIBBPF_OPTS` is a libbpf-specific macro for working with the libbpf userspace API, it's not a BPF feature.
